ServerCore/Pages/Responses/CreateBulkMulti.cshtml.cs

```diff
@@ -0,0 +1,199 @@
+                    {
+                        ModelState.AddModelError("IsSolution", "Unmatched IsSolution without Puzzle");
+                    }
+                    if (note != null)
+                    {
+                        ModelState.AddModelError("Note", "Unmatched Note without Puzzle");
+                    }
+
+                    // we're done
+                    break;
+                }
+
+                if (submittedText == null)
+                {
+                    ModelState.AddModelError("SubmittedText", "Unmatched Puzzle without Submission");
+                    break;
+                }
+
+                string submittedTextFormatted = ServerCore.DataModel.Response.FormatSubmission(submittedText);
+
+                // Ensure that the submission text is unique for this puzzle.
+                async Task<bool> isSubmittedTextUnique(string text)
+                {
+                    bool duplicateResponse = await(from Response r in _context.Responses
+                                                   where r.PuzzleID == puzzleId &&
+                                                   r.SubmittedText == text
+                                                   select r).AnyAsync();
+
+                    if (duplicateResponse)
+                    {
+                        ModelState.AddModelError("SubmittedText", "Submission text is not unique");
+                    }
+
+                    return !duplicateResponse;
+                };
+
+                if (!DeleteExisting && !await isSubmittedTextUnique(submittedTextFormatted))
+                {
+                    break;
+                }
+
+                if (!submissionsLookup[puzzleName].Add(submittedTextFormatted))
+                {
+                    ModelState.AddModelError("SubmittedText", "Submission text is not unique");
+                    break;
+                }
+
+                if (responseText == null)
+                {
+                    ModelState.AddModelError("SubmittedText", "Unmatched Puzzle without Response");
+                    break;
+                }
+
+                isSolution = isSolution == null ? string.Empty : isSolution.ToLower();
+
+                if (puzzleId == -1)
+                {
+                    ModelState.AddModelError("PuzzleName", "Bug in puzzleId lookup");
+                    break;
+                }
+
+                Response response = new Response()
+                {
+                    PuzzleID = puzzleId,
+                    SubmittedText = submittedText,
+                    ResponseText = responseText,
+                    Note = note,
+                    IsSolution = isSolution == "y" || isSolution == "yes" || isSolution == "t" || isSolution == "true" || isSolution == "1"
+                };
+
+                _context.Responses.Add(response);
+            }
+
+            if (!ModelState.IsValid)
+            {
+                return OnGet();
```

Comment:
Same question about Page()

Reply:
The existing non-multi versions return await OnGetAsync(puzzleId) and I was keeping the same pattern as those. If all four bulk pages can return Page() I can fix them all but I'm not super clear on the difference.